backend/src/v1/reservations/reservations.service.ts

```diff
@@ -187,6 +187,124 @@ export const cancel = async (reservationId: number): Promise<void> => {
   }
 };
 
+type TransactionExecuteQueryType = (queryText: string, values?: any[]) => Promise<any>;
+
+/**
+ * 만료된 예약을 처리 후, 다음 예약자에게 예약을 할당하기 위한 예약 정보를 반환합니다.
+ * - 책이 할당 될때, status는 여전히 0이지만 endAt과 bookId가 null에서 각각 값이 할당됩니다.
+ * - 따라서 status가 0이면서 책이 할당 되어 bookId와 endAt을 가졌으면서 endAt이 현재 날짜보다 이전인 예약을 찾습니다.
+ * - status를 3을 부여하여 만료된 예약임을 표시합니다.
+ * @param transactionExecuteQuery 트랜잭션 처리를 위한 executeQuery
+ */
+const handleReservationOverdue = async (transactionExecuteQuery: TransactionExecuteQueryType) => {
+  const overDueReservations: {
+    slack: string;
+    title: string;
+    bookId: number;
+    bookInfoId: number;
+  }[] = await transactionExecuteQuery(`
+    SELECT
+      user.slack AS slack,
+      book_info.title AS title,
+      reservation.bookId AS bookId,
+      reservation.bookInfoId AS bookInfoId
+    FROM
+      reservation
+    LEFT JOIN user ON
+      user.id = reservation.userId
+    LEFT JOIN book_info ON
+      book_info.id = reservation.bookInfoId
+    WHERE
+      reservation.status = 0 AND
+      bookId IS NOT NULL AND
+      IFNULL(DATEDIFF(CURDATE(), DATE(reservation.endAt)), 0) >= 1
+    FOR UPDATE;
+    UPDATE
+      reservation
+    SET
+      status = 3
+    WHERE
+      reservation_id IN (
+        SELECT
+          reservation_id
+        FROM
+          reservation
+        WHERE
+          reservation.status = 0 AND
+          bookId IS NOT NULL AND
+          IFNULL(DATEDIFF(CURDATE(), DATE(reservation.endAt)), 0) >= 1
+      );
+  `);
```

Comment:
- `reservation_id` 대신 `reservation.id`를 사용해야됩니다.
- 서브쿼리 실행할 때 예약이 만료된 데이터의 id값을 가져오는 서브쿼리의 결과에 `expiredReservations`라는 별칭을 주어 모호성을 없앴습니다.

```suggestion
     WHERE reservation.id IN (
    SELECT id
      FROM (
        SELECT reservation.id
          FROM reservation
         WHERE reservation.status = 0
               AND bookId IS NOT NULL
               AND IFNULL(DATEDIFF(CURDATE(), DATE(reservation.endAt)), 0) >= 1
           ) AS expiredReservations
       );
  `);
```